addons/l10n_in_website_sale/controllers/main.py

```diff
@@ -0,0 +1,35 @@
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.http import request
+
+
+class WebsiteSaleInherit(WebsiteSale):
+
+    def _validate_address_values(
+        self,
+        address_values,
+        partner_sudo,
+        address_type,
+        use_delivery_as_billing,
+        required_fields,
+        is_main_address,
+        **_kwargs,
+    ):
+        invalid_fields, missing_fields, error_messages = super()._validate_address_values(
+            address_values,
+            partner_sudo,
+            address_type,
+            use_delivery_as_billing,
+            required_fields,
+            is_main_address,
+            **_kwargs,
+        )
+        if not (country_id := address_values.get('country_id')):
+            return invalid_fields, missing_fields, error_messages
+        country = request.env['res.country'].search([('id', '=', country_id)], limit=1)
```

Comment:
```suggestion
        country = self.env['res.country'].search([('id', '=', country_id)], limit=1)
```
In master now you can access `self.env` in controller